src/ui/projectmanager/pms/pm.go

```diff
@@ -0,0 +1,410 @@
+	if len(value) != 0 {
+		enable, err := strconv.ParseBool(value)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse __enableContentTrust %s to bool: %v", value, err)
+		}
+		project.EnableContentTrust = enable
+	}
+
+	value = p.CustomProperties["__preventVulnerableImagesFromRunning"]
+	if len(value) != 0 {
+		prevent, err := strconv.ParseBool(value)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse __preventVulnerableImagesFromRunning %s to bool: %v", value, err)
+		}
+		project.PreventVulnerableImagesFromRunning = prevent
+	}
+
+	value = p.CustomProperties["__preventVulnerableImagesFromRunningSeverity"]
+	if len(value) != 0 {
+		project.PreventVulnerableImagesFromRunningSeverity = value
+	}
+
+	value = p.CustomProperties["__automaticallyScanImagesOnPush"]
+	if len(value) != 0 {
+		scan, err := strconv.ParseBool(value)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse __automaticallyScanImagesOnPush %s to bool: %v", value, err)
+		}
+		project.AutomaticallyScanImagesOnPush = scan
+	}
+
+	return project, nil
+}
+
+// IsPublic ...
+func (p *ProjectManager) IsPublic(projectIDOrName interface{}) (bool, error) {
+	project, err := p.get(projectIDOrName)
+	if err != nil {
+		return false, err
+	}
+	if project == nil {
+		return false, nil
+	}
+
+	return project.Public, nil
+}
+
+// Exist ...
+func (p *ProjectManager) Exist(projectIDOrName interface{}) (bool, error) {
+	project, err := p.get(projectIDOrName)
+	if err != nil {
+		return false, err
+	}
+
+	return project != nil, nil
+}
+
+// GetRoles ...
+func (p *ProjectManager) GetRoles(username string, projectIDOrName interface{}) ([]int, error) {
+	if len(username) == 0 || projectIDOrName == nil {
+		return nil, nil
+	}
+
+	id, err := p.getIDbyHarborIDOrName(projectIDOrName)
+	if err != nil {
+		return nil, err
+	}
+
+	// get expanded project which contains role info by GET /projects/id?expand=true
+	path := fmt.Sprintf("/projects/%s?expand=true", id)
+	data, err := p.send(http.MethodGet, path, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	pro := &project{}
+	if err = json.Unmarshal(data, pro); err != nil {
+		return nil, err
+	}
+
+	roles := []int{}
+
+	for _, user := range pro.Administrators {
+		if user.Email == username {
+			roles = append(roles, common.RoleProjectAdmin)
+			break
+		}
+	}
+
+	for _, user := range pro.Developers {
+		if user.Email == username {
+			roles = append(roles, common.RoleDeveloper)
+			break
+		}
+	}
+
+	for _, user := range pro.Guests {
+		if user.Email == username {
+			roles = append(roles, common.RoleGuest)
+			break
+		}
+	}
+
+	return roles, nil
+}
+
+func (p *ProjectManager) getIDbyHarborIDOrName(projectIDOrName interface{}) (string, error) {
+	pro, err := p.get(projectIDOrName)
+	if err != nil {
+		return "", err
+	}
+
+	if pro == nil {
+		return "", fmt.Errorf("project %v not found", projectIDOrName)
+	}
+
+	return pro.ID, nil
+}
+
+// GetPublic ...
+func (p *ProjectManager) GetPublic() ([]*models.Project, error) {
```

Comment:
reuse get?
the func get should take a map and query with "/projects?filter= $k eq $v"